src/rpc/rawtransaction.cpp

```diff
@@ -573,7 +573,6 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
             + HelpExampleRpc("createrawtransaction", "\"[{\\\"txid\\\":\\\"myid\\\",\\\"vout\\\":0}]\", \"{\\\"address\\\":0.01}\"")
         );
 
-    LOCK(cs_main);
```

Comment:
This kind of deletion is one that *really* makes me want to get rid of globals. I've traced through the calls following this and I'm pretty sure that there's nothing that needs this lock. I guess that the worst thing that can happen is that the user gets returned a raw transaction that's invalid after a network upgrade, but that seems fine.

Reply:
I love the way you're thinking -- Bitcoin Core doesn't do network upgrades, so a change might affect us in ways that aren't relevant to them. I think in this case it's okay, because even with the `LOCK()`, as soon as the lock is released (but even before the RPC returns), there could be a NU that makes the transaction invalid. The same window exists either way. But also, I think the tx construction changes a couple of blocks before a NU activates.

Reply:
We can't remove this entirely, because we added a call to `chainActive.Height()` in order to create the right version of transaction for the current network epoch.

Instead, modify this commit to reduce the scope of this lock (rather than removing it completely):
```diff
diff --git a/src/rpc/rawtransaction.cpp b/src/rpc/rawtransaction.cpp
index bcfeb0ffbd..d1ef396a8e 100644
--- a/src/rpc/rawtransaction.cpp
+++ b/src/rpc/rawtransaction.cpp
@@ -573,7 +573,6 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
             + HelpExampleRpc("createrawtransaction", "\"[{\\\"txid\\\":\\\"myid\\\",\\\"vout\\\":0}]\", \"{\\\"address\\\":0.01}\"")
         );
 
-    LOCK(cs_main);
     RPCTypeCheck(params, boost::assign::list_of(UniValue::VARR)(UniValue::VOBJ)(UniValue::VNUM)(UniValue::VNUM), true);
     if (params[0].isNull() || params[1].isNull())
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, arguments 1 and 2 must be non-null");
@@ -581,7 +580,11 @@ UniValue createrawtransaction(const UniValue& params, bool fHelp)
     UniValue inputs = params[0].get_array();
     UniValue sendTo = params[1].get_obj();
 
-    int nextBlockHeight = chainActive.Height() + 1;
+    int nextBlockHeight;
+    {
+        LOCK(cs_main);
+        nextBlockHeight = chainActive.Height() + 1;
+    }
     CMutableTransaction rawTx = CreateNewContextualCMutableTransaction(
         Params().GetConsensus(), nextBlockHeight);
 
```